web/packages/teleterm/src/services/pty/ptyHost/buildPtyOptions.ts

```diff
@@ -46,9 +52,28 @@ export async function buildPtyOptions(
   cmd: PtyCommand
 ): Promise<{
   processOptions: PtyProcessOptions;
+  openedShell: Shell;
   creationStatus: PtyProcessCreationStatus;
 }> {
-  return resolveShellEnvCached(settings.defaultShell)
+  // Get the full shell object by the shell ID.
+  // We wouldn't have to do it if we sent a shell bin path from the renderer,
+  // but it's better to validate inputs that come from that process.
+  const defaultShell = settings.availableShells.find(
+    s => s.id === settings.defaultOsShellId
+  );
+  let shell = defaultShell;
+  let failedToResolveShell = false;
+
+  if (cmd.kind === 'pty.shell') {
+    const resolvedShell = await resolveShell(cmd, settings, options);
+    if (!resolvedShell) {
+      failedToResolveShell = true;
+    } else {
+      shell = resolvedShell;
+    }
+  }
+
+  return resolveShellEnvCached(shell.binPath)
```

Comment:
This function works only on Unixes. How does this all behave if I have custom env vars set up in WSL?

Reply:
On Windows we don't have to resolve the shell env, because it always sets the correct env vars for the process.
However, there is one thing that I missed. To pass variables to WSL, they need to be defined in `WSLENV`. Unfortunately, I overwrite `WSLENV` with our variables in `buildPtyOptions`, so if the user has its own value there, it will be ignored. To fix this, we only need to prepend our WSLENV to the existing one.

I tested that by specifying a custom WSLENV in Windows environment variables:
<img width="545" alt="image" src="https://github.com/user-attachments/assets/14804c25-e3ba-4ad8-8d7b-834217e3b8b1">

It is now passed correctly:
<img width="769" alt="image" src="https://github.com/user-attachments/assets/3e5cdc07-98f7-42c4-a218-3e687526f003">